scripts/commands/release.py

```diff
@@ -0,0 +1,129 @@
+# Make a release commit on o-spreadsheet versions
+# This will be handled by a Github action to automatically publish the release on npm
+
+import os
+import re
+import json
+import configparser
+import subprocess
+from datetime import date
+from uuid import uuid4
+from helpers import (
+    checkout,
+    get_commits,
+    fetch_repositories,
+    get_o_spreadsheet_release_prs,
+    make_PR,
+    reset,
+    spreadsheet_release_title,
+)
+from utils import pushd, get_o_spreadsheet_hash
+from shared import spreadsheet_odoo_versions
+
+
+def release(config: configparser.ConfigParser):
+    # todo
+    print("\n=== RELEASE O-SPREADSHEET ===\nThis may take a while ;-)\n")
+    spreadsheet_path = config["spreadsheet"]["repo_path"]
+    fetch_repositories(config)
+    old_prs = []
+    new_prs = []
+    existing_prs = get_o_spreadsheet_release_prs(config)
+    today = date.today()
+    d = f"{str(today.day).zfill(2)}{str(today.month).zfill(2)}"
+    h = str(uuid4())[:4]
+    for [repo, version, rel_path] in spreadsheet_odoo_versions.values():
+        text = f"Processing version {version}"
+        print(text)
+        print("=" * len(text))
+        if version in existing_prs:
+            print(
+                f"Branch {version} already has a pending release PR on odoo/o-spreadsheet. Skipping..."
+            )
+            old_prs.append([version, existing_prs[version]])
+            continue
+
+        repo_path = config[repo]["repo_path"]
+        full_path = os.path.join(repo_path, rel_path)
+
+        # checkout o-spreadsheet
+        checkout(spreadsheet_path, version, force=True)
+        reset(spreadsheet_path, version)
+
+        # checkout Odoo  on update branch
+        checkout(repo_path, version, force=True)
+        reset(repo_path, version)
```

Comment:
odoo ? This is useless, right ?

Reply:
the idea is not to release if theere is no diff in Odoo. Otherwise, we could start release 10 times and make 10 empty release commits (ok it wouldn't work because we don't allow empty commits but this seems better than waiting for an exit(1) from git commit.

Reply:
won't it hit `f"Branch {version} is up-to-date on odoo/{repo}. Skipping..."` just below if there's no diff (=no commit ?) ?